db/migrate/20230901144301_add_links_json_to_affiliates.rb

```diff
@@ -0,0 +1,5 @@
+class AddLinksJsonToAffiliates < ActiveRecord::Migration[7.0]
+  def change
+    add_column :affiliates, :links_json, :json
```

Comment:
Why are we opting for using `json` to store this data instead of a table? Is there an advantage or something we need to be compatible with driving this decision?

My suggestion would be to do it more "the rails" way. We could have a `Link` model with all the related validation logic in there, It'd need to be a polymorphic relationship. I'll be happy to help modeling it out.